inference-engine/src/cldnn_engine/cldnn_config.cpp

```diff
@@ -253,9 +297,22 @@ void Config::UpdateFromMap(const std::map<std::string, std::string>& configMap)
         } else {
             IE_THROW(NotFound) << "Unsupported property key by plugin: " << key;
         }
-
-        adjustKeyMapValues();
     }
+#if (IE_THREAD == IE_THREAD_TBB || IE_THREAD == IE_THREAD_TBB_AUTO)
+    if (prev_cpu_core_type != cpu_core_type
+        || prev_n_threads != n_threads) {
+            if (getAvailableCoresTypes().size() > 1) {
+                if (cpu_core_type == IStreamsExecutor::Config::BIG
+                    || cpu_core_type == IStreamsExecutor::Config::LITTLE) {
+                        n_threads = std::min(n_threads, static_cast<size_t>(getNumberOfCores(cpu_core_type)));
+                    }
+            } else {
+                cpu_core_type = IStreamsExecutor::Config::ANY;
+                n_threads = std::min(n_threads, static_cast<size_t>(std::thread::hardware_concurrency()));
```

Comment:
why do we need "prev_xxxx " ? why not just update within the conditions...?

Reply:
becuase Config::UpdateFromMap is called three times for iteration, I need to check the data with prev_xxx so that reduce duplicated operation.

Reply:
Modified code to remove duplicated ops. please review it again.